src/rfkill/rfkill.c

```diff
@@ -257,35 +268,93 @@ static int load_state(
         return 0;
 }
 
-static int save_state(
+static void save_state_queue_remove(int idx, char *state_file) {
+
+        struct write_queue_item *item, *tmp;
+
+        LIST_FOREACH_SAFE(queue, item, tmp, write_queue) {
+                if ((state_file && strcmp(item->file, state_file) == 0) || idx == item->rfkill_idx) {
+                        log_debug("Canceled previous save state of '%s' to %s.", one_zero(item->state), item->file);
+                        LIST_REMOVE(queue, write_queue, item);
+                        free(item->file);
+                        mfree(item);
```

Comment:
no need to use mfree() if you don't care about its return value... the whole reason mfree() exists, is so that you can write: 

```c
p = mfree(p);
```

and that frees `p` and also assigns NULL to it...